### ISSUE TITLE
[BUG] Model does not raises error when convergence with MAP fails

### ISSUE THREAD

**Issue by felipeangelimvieira** [felipeangelimvieira/prophetverse#39]
Some hyperparameters may cause unstable optimization and lead to nans. The code currently ignores this and return the nans without raising the error. An exception with informative message (suggesting the change of learning rate for example) could  improve user experience.

**Comment by felipeangelimvieira** [comment]
Using `numpyro.enable_x64` solves the NaNs. Need to check if it affects performance.
